### PR TITLE
Update home to be lazy loaded

### PR DESCRIPTION
**Pull request by aminpaks**

### CODE REVIEW

**Comment by nasreddineskandrani** [comment]
we need to add a prefix to each commit at least for now `feat:` or `fix:` or `tool:`

File: src/App.tsx [hunk 1]
```diff
@@ -1,15 +1,21 @@
-import React, { Component } from 'react';
-import { BrowserRouter, Route } from 'react-router-dom';
+import React, { lazy, Suspense, Component } from 'react';
+import { BrowserRouter, Route, Switch } from 'react-router-dom';
 import { GlobalStyles } from './App.styled';
-import { HomePage } from './Pages';
+
+const HomePage = lazy(() => import('./Pages/Home'));
```
Comment (nasreddineskandrani) [inline src/App.tsx]: this should come from the folder `Home` all what you need here is
import `LazyHomePage`
and render it

File: src/App.tsx [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
+const renderHomePage = (props: any) => <HomePage {...props} />;
```
Comment (nasreddineskandrani) [inline src/App.tsx]: we need to enable `noImplicitAny` compiler option and `noAny` ts lint rule :)

**Comment by nasreddineskandrani** [comment]
the PR should merge to `master`*

File: src/App.tsx (hunk 1 above)
Comment (nasreddineskandrani) [reply src/App.tsx]: when i say from `/home` i mean you can split `Home` to
/home/lazy
/home/none-lazy

And in the lazy you boot the none-lazy

OR

we can create a file lazy.js in the root of pages and add all lazy stuf inside.

Comment (aminpaks) [reply src/App.tsx]: I guess

**Comment by aminpaks** [comment]
No, if PRs are sequential they should be merge into each other and the last one can be merge to the main branch. I think I should've mention where we should start merging.

File: src/App.tsx (hunk 2 above)
Comment (aminpaks) [reply src/App.tsx]: `noImplicitAny` is already on. And annotating the code here is not useful for anything.

**Comment by aminpaks** [comment]
Comments addressed.

File: src/Pages/Home/Route.tsx [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import React, { lazy } from 'react';
+import { Route } from 'react-router-dom';
+
+const HomePage = lazy(() => import('./Home'));
+export const renderHomePage = (props: any) => <HomePage {...props} />;
```
Comment (aminpaks) [inline src/Pages/Home/Route.tsx]: This line is not needed at all, we should be doing:
```
const HomePage = lazy(() => import('./Home'));
export const HomePageRoute = () => <Route path="/" component={HomePage} />
```

But lazy is very new and the new route version is on the way.

Comment (nasreddineskandrani) [reply src/Pages/Home/Route.tsx]: when you do import('./Home') and Route is inside it doesn't break the chunk?

Comment (aminpaks) [reply src/Pages/Home/Route.tsx]: So yes. `Route.tsx` will be part of main bundle and the rest is lazy loaded.

File: src/App.tsx (hunk 2 above)
Comment (nasreddineskandrani) [reply src/App.tsx]: oki

**Pull request merged**